### PR TITLE
Added docker file to simplify development

### PR DESCRIPTION
**Pull request by mikefarah**
## Description
I want to add an enhancement to this plugin to allow the response status to be set (we have a scenario where the status is in the body...)

I struggled quite a bit with trying various lua/openssl version combinations so that I could successfully run:
`make setup`
`make install`
`make test`

To that end, I've created a docker file that includes all the right version to get those commands to pass.

## How Has This Been Tested?
Built the docker image, volume mounted the source and successfully ran the various make commands.

## Checklist

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  luarocks \
+  git \
+  libssl1.0-dev \
+  make
+
+RUN luarocks install --server=http://luarocks.org/dev luaffi
+ADD Makefile .
+RUN make setup
+
+ADD kong-plugin-template-transformer-0.8.0-2.rockspec .
```
Comment (PFadel) [inline Dockerfile]: Could we use something like

```suggestion
ADD kong-plugin-template-transformer-*.rockspec .
```

Just so we don't need to change the Dockerfile every release.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
+FROM debian
+
+RUN apt-get update
+
+RUN apt-get install -y \
+  lua5.2 \
```
Comment (PFadel) [inline Dockerfile]: In development we always used Lua 5.1.5, this didn't work for you ?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  liblua5.2-dev \
+  luarocks \
+  git \
+  libssl1.0-dev \
+  make
+
+RUN luarocks install --server=http://luarocks.org/dev luaffi
```
Comment (PFadel) [inline Dockerfile]: Why do we need this ? If it's a dependency we should have listed in Makefile.

File: Dockerfile (hunk 2 above)
Comment (PFadel) [reply Dockerfile]: Actually I can't remember the reason so if it worked I guess it's ok.

File: Dockerfile (hunk 3 above)
Comment (PFadel) [reply Dockerfile]: Unfortunately my lua knowledge isn't much better. But it really does not work without it.

If you could then please add it to the Makefile I would be happy to approve.

**Comment by PFadel** [comment]
@mikefarah after this last commit i'm not able to build the Dockerfile. I got the following error:

```bash
--server=http://luarocks.org/dev luaffi scm-1 not found, installing via luarocks...
Installing http://luarocks.org/dev/luaffi-scm-1.src.rock

Error: Could not fetch rock file: Error fetching file: Failed downloading http://luarocks.org/dev/luaffi-scm-1.src.rock - luaffi-scm-1.src.rock
```

**Comment by mikefarah** [comment]
Worked for me 🤔 

```
--server=http://luarocks.org/dev luaffi scm-1 not found, installing via luarocks...
Installing http://luarocks.org/dev/luaffi-scm-1.src.rock
gcc -O2 -fPIC -I/usr/include/lua5.2 -c call.c -o call.o -Idynasm
gcc -O2 -fPIC -I/usr/include/lua5.2 -c ctype.c -o ctype.o -Idynasm
gcc -O2 -fPIC -I/usr/include/lua5.2 -c ffi.c -o ffi.o -Idynasm
gcc -O2 -fPIC -I/usr/include/lua5.2 -c parser.c -o parser.o -Idynasm
gcc -shared -o ffi.so -L/usr/local/lib call.o ctype.o ffi.o parser.o
luaffi scm-1 is now installed in /usr/local (license: BSD)
```

**Comment by biancarosa** [comment]
Awesome! It works here @PFadel, inside and outside of the container.

**Comment by PFadel** [comment]
:woman_shrugging: probable something wrong with my docker, then.

**Pull request merged**